### PR TITLE
configurePlugins command for tsserver

### PR DESCRIPTION
**Pull request by minestarks**
Implements #18604

- Add a new tsserver command, `configurePlugins`.
- Add a new optional method to `PluginModule`, `onConfigurationChanged`. Plugin authors can implement this.
- When `configurePlugins` is invoked, it will pass on the configuration object to any loaded plugin instances with a  matching name.
- This configuration is meant to override the initial configuration passed in in `create`/`PluginCreateInfo` as well as any previous configurations passed in in `configurePlugins`.
- The updated configuration will also be remembered by tsserver so that it can be passed in, in the `create` call, to any projects loaded in the future with the matching plugin. This is needed for consistency -- the caller of `configurePlugins` would have no way of knowing whether the plugin/project of interest has yet been loaded.

cc @mjbvz

### CODE REVIEW

**Comment by minestarks** [comment]
@mjbvz Thanks! Just opened corresponding PR https://github.com/Microsoft/vscode/pull/61756 , let me know if that works.

File: src/server/project.ts [hunk 1]
```diff
@@ -1180,12 +1190,25 @@ namespace ts.server {
...
                 this.languageService = newLS;
-                this.plugins.push(pluginModule);
+                this.plugins.push({ name: configEntry.name, module: pluginModule });
             }
             catch (e) {
                 this.projectService.logger.info(`Plugin activation failed: ${e}`);
             }
         }
+
+        onPluginConfigurationChanged(pluginName: string, configuration: any) {
```
Comment (sheetalkamat) [inline src/server/project.ts]: Are you sure this needs to be exposed and not internal?

**Comment by sheetalkamat** [comment]
Changes look good but we need test.

File: src/server/project.ts (hunk 1 above)
Comment (minestarks) [reply src/server/project.ts]: Thanks I'll make it internal. Wasn't sure about the logic behind what's internal and what's public.

**Comment by minestarks** [comment]
@sheetalkamat test added!

**Pull request merged**